File: src/main/java/com/refinedmods/refinedstorage/api/storage/disk/IStorageDiskRegistry.java
package com.refinedmods.refinedstorage.api.storage.disk;

import net.minecraft.util.ResourceLocation;

import javax.annotation.Nullable;

/**
 * Stores factories for reproducing storage disks from disk.
 */
public interface IStorageDiskRegistry {
    /**
     * Adds a factory.
     *
     * @param id      the id of this factory
     * @param factory the factory
     */
    void add(ResourceLocation id, IStorageDiskFactory<?> factory);

    /**
     * Gets a factory.
     *
     * @param id the factory id
     * @return the factory, or null if no factory was found
     */
    @Nullable
    IStorageDiskFactory get(ResourceLocation id);
}

File: src/main/kotlin/com/refinedmods/refinedstorage/api/storage/disk/IStorageDiskRegistry.kt
package com.refinedmods.refinedstorage.api.storage.disk

import net.minecraft.util.Identifier


/**
 * Stores factories for reproducing storage disks from disk.
 */
interface IStorageDiskRegistry {
    /**
     * Adds a factory.
     *
     * @param id      the id of this factory
     * @param factory the factory
     */
    fun add(id: Identifier?, factory: IStorageDiskFactory<*>?)

    /**
     * Gets a factory.
     *
     * @param id the factory id
     * @return the factory, or null if no factory was found
     */
    operator fun get(id: Identifier): IStorageDiskFactory<*>?
}